utils/histogram.hh

```diff
@@ -118,9 +118,13 @@ public:
      * Return true if the current event should be sample.
      * In the typical case, there is no need to use this method
      * Call set_latency, that would start a latency object if needed.
+     *
+     * Typically, sample_mask is of the form of 2^n-1 which would
+     * mean that we sample one of 2^n, but setting sample_mask to zero
+     * would mean we would always sample.
      */
-    bool should_sample() const {
-        return total == 0 || (started & sample_mask);
+    bool should_sample() const noexcept {
+        return total == 0 || ((started & sample_mask) == sample_mask);
```

Comment:
Maybe deserves a bug # and backports. What's the impact?

This sample_mask is very unintuitive, we should replace it (but ok after the series).

Reply:
I think @tgrabiec found it years ago, I couldn't find an issue, I'm planning to deprecate this class alltogether but currently, this bug prevent disable the sampling.

Currently, what the code does is sample half of the points instead of according to the mask (1 out of 1^n).

Reply:
https://github.com/scylladb/scylla/issues/2747